src/cloudflare/sockets.ts

```diff
@@ -10,3 +10,11 @@ export function connect(
 ): sockets.Socket {
   return sockets.connect(address, options);
 }
+
+interface Fetcher {
+  fetch: typeof fetch;
+}
+
+export function convertSocketToFetcher(socket: sockets.Socket): Fetcher {
```

Comment:
This feels non-idiomatic as a public API.

Something like `new Fetcher(socket)` would be idiomatic -- except I don't think we actually expose `Fetcher` as a concrete JS class anywhere, and if we decided to expose it, maybe we want to bikeshed the name first...

If we keep this as a freestanding function, maybe it could be called `newHttpClient()`.